synapse/handlers/room.py

```diff
@@ -645,6 +645,13 @@ def create_room(self, requester, config, ratelimit=True, creator_join_profile=No
                 check_membership=False,
             )
 
+        if is_public:
+            if not self.config.is_alias_creation_allowed(user_id, room_id, room_alias):
```

Comment:
I believe this should be calling `is_publishing_room_allowed`, not `is_alias_creation_allowed`.

Reply:
Yes you are correct. My mistake! I have the same rule for both so I didn't notice while testing it. I pushed a fix.